src/toolbox.ts

```diff
@@ -2456,35 +2457,40 @@ export class SubmitButtons extends ToolboxItem {
         }
     }
 
+    /**
+     * Group submit buttons by row number
+     * 
+     * @returns {ULabelSubmitButtons[]} Array of submit buttons grouped by row number
+     */
+    private sort_buttons_by_row_number() {
+        let submit_buttons_by_row: ULabelSubmitButtons[] = [];
+        // First, get all the unique row numbers. 
+        // If a button doesn't have a row number, it will be placed in row 0.
+        let row_numbers: Set<number> = new Set(this.submit_buttons.map((button) => button.row_number ? button.row_number : 0));
+        // Sort the row numbers
+        let sorted_row_numbers: number[] = Array.from(row_numbers).sort((a, b) => a - b);
+        // Group the buttons by row number in ascending order
+        for (let row_number of sorted_row_numbers) {
+            submit_buttons_by_row.push(
+                this.submit_buttons.filter((button) => {
+                    // If the button doesn't have a row number, it will be placed in row 0
+                    if (button.row_number === undefined) {
+                        return row_number === 0;
+                    }
```

Comment:
Is this the right spot for this? Can the `ULabelSubmitButton` just have a default value of 0?

Reply:
i dont think types can have defaults, i could make `SubmitButton` an actual class i suppose.

Reply:
Yeah it would have to be a manual check on init or similar. Probably fine for now but if this prop gets used in other places it might start to make sense